### PR TITLE
added timeout & fetch defaults files

### PR DESCRIPTION
**Pull request by roee-hershko**

### CODE REVIEW

File: src/utils/defaults.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module.exports.fetchOptions = {
+  timeout: 10000,
```
Comment (jeeng) [inline src/utils/defaults.js]: try to retrieve a timeout also from `process.env.JTA_FETCH_TIMEOUT` and default to 10000 is missing

File: src/utils/fetch.js [hunk 2]
```diff
@@ -37,6 +38,10 @@ module.exports = (url, options = {}) => {
       })
       .on("error", err => reject(err));
 
+      req.setTimeout((opts.timeout), () => {
```
Comment (jeeng) [inline src/utils/fetch.js]: What happens on `req.abort()`?
You should also refactor `Promise.race`

Comment (roee-hershko) [reply src/utils/fetch.js]: there is 2 kind of timeouts on the native http module
1. options.timeout = pre handshake timeout
2. req.setTimeout = request flight timeout

if connection esatblished, req.timeout is ignored and req.setTimeout is required, without it the requests can connection will stay open

File: src/utils/defaults.js (hunk 1 above)
Comment (roee-hershko) [reply src/utils/defaults.js]: done

**Pull request merged**